src/application.coffee

```diff
@@ -578,11 +578,13 @@ application.update = update = (force) ->
 			throw new Error(joinErrorMessages(failures)) if failures.length > 0
 		.then ->
 			updateStatus.failed = 0
+			device.setUpdatePending(false)
 			# We cleanup here as we want a point when we have a consistent apps/images state, rather than potentially at a
 			# point where we might clean up an image we still want.
 			dockerUtils.cleanupContainersAndImages()
 		.catch (err) ->
 			updateStatus.failed++
+			device.setUpdatePending(true)
```

Comment:
The use case for this is a user opting in for an update when the update lock is set. So the supervisor will download the update, try to apply it and fail, then set this flag so the user knows there's an update pending installation.